airflow/models.py

```diff
@@ -826,12 +832,16 @@ def are_dependencies_met(
                 TI.state == State.SUCCESS,
             ).first()
             if not previous_ti:
+                if verbose:
+                    logging.warning("depends_on_past not satisfied")
                 return False
 
             # Applying wait_for_downstream
             previous_ti.task = self.task
             if task.wait_for_downstream and not \
                     previous_ti.are_dependents_done(session):
+                if verbose:
+                    logging.warning("wait_for_downstream not satisfied")
```

Comment:
Looks better. is it possible at this stage to get even more information e.g. which of (possibly many) dependencies failed.
BTW. I got confused - isn't it upstream task that is a dependency here, not a downstream one?